core-api/src/telemetry/metrics.rs

```diff
@@ -0,0 +1,280 @@
+use std::{borrow::Cow, collections::HashSet, fmt::Debug, sync::Arc};
+
+/// Implementors of this trait are expected to be defined in each language's bridge.
+/// The implementor is responsible for the allocation/instantiation of new metric meters which
+/// Core has requested.
+pub trait CoreMeter: Send + Sync + Debug {
+    fn new_attributes(&self, attribs: MetricsAttributesOptions) -> MetricAttributes;
+    fn counter(&self, params: MetricParameters) -> Arc<dyn Counter>;
+    fn histogram(&self, params: MetricParameters) -> Arc<dyn Histogram>;
+    fn gauge(&self, params: MetricParameters) -> Arc<dyn Gauge>;
+}
+
+#[derive(Debug, Clone, derive_builder::Builder)]
+pub struct MetricParameters {
+    /// The name for the new metric/instrument
+    #[builder(setter(into))]
+    pub name: Cow<'static, str>,
+    /// A description that will appear in metadata if the backend supports it
+    #[builder(setter(into), default = "\"\".into()")]
+    pub description: Cow<'static, str>,
+    /// Unit information that will appear in metadata if the backend supports it
+    #[builder(setter(into), default = "\"\".into()")]
+    pub unit: Cow<'static, str>,
```

Comment:
Can we document the unit string core uses for its durations? We will have to document this lang side so that when users use durations or when users write a backend, they know to match (though lang side is probably going to be more strongly typed than Rust here I suspect, which is a weird statement).